### PR TITLE
Makefile support OS X. Additional tests for node. Remove unused request interfaces

### PR DESCRIPTION
**Pull request by MaxFedotov**
Hi @prometherion, 

Thanks a lot for your work :)
When I've tried to run `make e2e` on OS X found an issue with mkcert rootCA location - it is different then on Linux, so I've fixed it.
Also I've added tests for cordon\taint (although they are just the same editing node spec, but in case something changes in future I think it is a good idea to have them)
And I removed unused interface methods from request

### CODE REVIEW

File: e2e/run.bash [hunk 1]
```diff
@@ -9,4 +9,4 @@ echo ">>> Waiting for capsule pod to be ready for accepting requests"
 kubectl --namespace capsule-system wait --for=condition=ready --timeout=320s pod -l app.kubernetes.io/instance=capsule
 
 echo ">>> Starting test suite"
-bats -t "$(git rev-parse --show-toplevel)"/e2e/tests/06_*
```
Comment (prometherion) [inline e2e/run.bash:12]: Oh, gosh: it's embarrassing, I forgot to remove this 🤣

**Review by prometherion — CHANGES_REQUESTED** [review]
Could you split changes in 3 separate commits using the Semantic Commit Message style guide?

File: e2e/run.bash (hunk 1 above)
Comment (MaxFedotov) [reply e2e/run.bash:12]: :)

> Could you split changes in 3 separate commits using the Semantic Commit Message style guide?

done :)

Comment (MaxFedotov) [reply e2e/run.bash:12]: :) i was wondering, if it this change was needed somehow, but checked previous version and understood that it was just a typo

**Comment by MaxFedotov** [comment]
@prometherion done :)

**Review by prometherion — APPROVED** [review]

File: e2e/tests/06_node_get.bats [hunk 2]
```diff
@@ -20,10 +18,45 @@ teardown() {
...
+  kubectl taint nodes capsule-control-plane key1=value1:NoSchedule" 3 5
+}
+
+@test "Node cordoning via kubectl" {
+  run sh -c "KUBECONFIG=${HACK_DIR}/alice.kubeconfig kubectl cordon capsule-control-plane"
+  [ $status -eq 1 ]
+
+  kubectl patch tenants.capsule.clastix.io node --type=json -p '[{"op": "add", "path": "/metadata/annotations", "value": {"capsule.clastix.io/enable-node-listing": "true", "capsule.clastix.io/enable-node-update": "true"}}]'
+  poll_until_equals "node cordoning" "node/capsule-control-plane cordoned" "KUBECONFIG=${HACK_DIR}/alice.kubeconfig
+  kubectl cordon capsule-control-plane" 3 5
```
Comment (prometherion) [inline e2e/tests/06_node_get.bats:51]: Hey, can we __uncordon__ after the execution? Just to restore the initial situation :)

**Review by prometherion — COMMENTED** [review]
I noticed a conflict, could you fix it, please?

File: e2e/tests/06_node_get.bats (hunk 2 above)
Comment (MaxFedotov) [reply e2e/tests/06_node_get.bats:53]: fixed

File: e2e/tests/06_node_get.bats [hunk 3]
```diff
@@ -20,10 +18,45 @@ teardown() {
...
+  kubectl patch tenants.capsule.clastix.io node --type=json -p '[{"op": "add", "path": "/metadata/annotations", "value": {"capsule.clastix.io/enable-node-listing": "true", "capsule.clastix.io/enable-node-update": "true"}}]'
   poll_until_equals "node labeling" "node/capsule-control-plane labeled" "KUBECONFIG=${HACK_DIR}/alice.kubeconfig kubectl label node capsule-control-plane capsule.clastix.io/test=labeling" 3 5
 }
+
+@test "Node tainting via kubectl" {
+  run sh -c "KUBECONFIG=${HACK_DIR}/alice.kubeconfig kubectl taint nodes capsule-control-plane key1=value1:NoSchedule"
+  [ $status -eq 1 ]
+
+  kubectl patch tenants.capsule.clastix.io node --type=json -p '[{"op": "add", "path": "/metadata/annotations", "value": {"capsule.clastix.io/enable-node-listing": "true", "capsule.clastix.io/enable-node-update": "true"}}]'
+  poll_until_equals "node tainting" "node/capsule-control-plane tainted" "KUBECONFIG=${HACK_DIR}/alice.kubeconfig
```
[the remark this answers is not in the document]
Comment (MaxFedotov) [reply e2e/tests/06_node_get.bats:43]: fixed

**Comment by MaxFedotov** [comment]
@prometherion

> I noticed a conflict, could you fix it, please?

done

**Pull request merged**